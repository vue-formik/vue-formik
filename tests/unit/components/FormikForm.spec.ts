import { beforeEach, describe, expect, test, vi } from "vitest";
import { mount } from "@vue/test-utils";
import { FormikForm, useFormik } from "@/index";

describe("FormikForm", () => {
  let formik: ReturnType<typeof useFormik>
  const onSubmitMock = vi.fn()
  beforeEach(() => {
    formik = useFormik({
      initialValues: {
        name: "",
      },
      onSubmit: onSubmitMock,
    })
  })
  test("renders form",  () => {
    const wrapper = mount(FormikForm, {
      props: {
        formik
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  test("renders form with children",  () => {
    const wrapper = mount(FormikForm, {
      props: {
        formik
      },
      slots: {
        default: "<input name='name' />"
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  test("submit button calls formik.handleSubmit",  () => {
    const wrapper = mount(FormikForm, {
      props: {
        formik
      },
      slots: {
        default: "<button type='submit'>Submit</button>"
      }
    })
    const button = wrapper.find("button")
    expect(onSubmitMock).not.toHaveBeenCalled()
    button.trigger("submit")
    expect(onSubmitMock).toHaveBeenCalledTimes(1)
    const calledWith = onSubmitMock.mock.calls[0]
    expect(calledWith).toMatchSnapshot()
  })
})
