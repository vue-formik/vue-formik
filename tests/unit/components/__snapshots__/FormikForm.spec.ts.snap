// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormikForm > renders form 1`] = `"<form></form>"`;

exports[`FormikForm > renders form with children 1`] = `"<form><input name="name"></form>"`;

exports[`FormikForm > submit button calls formik.handleSubmit 1`] = `
[
  {
    "name": "",
  },
  {
    "reset": [Function],
    "setErrors": [Function],
    "setSubmitting": [Function],
    "setValues": [Function],
  },
]
`;
